source/git/gitsource_unix.go

```diff
@@ -4,13 +4,41 @@ package git
 
 import (
 	"context"
+	"fmt"
+	"os"
 	"os/exec"
 	"syscall"
 	"time"
+
+	"github.com/docker/docker/pkg/reexec"
+	"golang.org/x/sys/unix"
 )
 
+func init() {
+	reexec.Register("reexec-git", reexecGit)
+}
+
+func reexecGit() {
+	// Need standard user umask for git process.
+	unix.Umask(0022)
+
+	// Reexec git command
+	cmd := exec.Command("git", os.Args[2:]...)
```

Comment:
why isn't this `1:` ?

Reply:
Because we pass `reexec-git git ...` and `exec.Command` makes a `LookPath` on the `name`.

Reply:
we should then either not send `git` as the second argument, or use it in the first parameter of `exec.Command`